crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -367,6 +374,20 @@ impl<'w, T: Component> DerefMut for ResMut<'w, T> {
     }
 }
 
+impl<'w, T: Component> AsRef<T> for ResMut<'w, T> {
+    #[inline]
+    fn as_ref(&self) -> &T {
+        &*self
```

Comment:
Ideally, I'd like consistency between the `as_ref` and `as_mut` implementaiton. 
So either 
`&*self` and `&mut *self`
or
`self.deref()` and `self.deref_mut()`

Reply:
Done!